tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -2849,8 +2859,14 @@ private:
          std::string(filename), std::string(dirname), std::string(treename), columnListWithoutSizeColumns, options});
 
       ::TDirectory::TContext ctxt;
+#ifdef R__HAS_ROOT7
+      int tagForTTree = 1; // Meaningless type, used to call the right constructor
+      auto newRDF = ROOT::Internal::RDF::CreateRDFFromTTree(tagForTTree, fullTreeName, filename,
+                                                            /*defaultColumns=*/columnListWithoutSizeColumns);
```

Comment:
note that Snapshot only has to return a `RInterface<RLoopManager>`, so I think most of the gymnastic here to avoid the dependency cycle between RInterface and RDataFrame can be removed: it should be simpler to construct a `RInterface<RLoopManager>` directly, leaving the `RDataFrame` type out of the game.

IIUC you could move `CreateLMFrom{Tree,NTuple}` to `RLoopManager.{hxx,cxx}` and call `CreateLMFromTree` directly from here without risking dependency cycles and without needing an extra mechanism to bypass the file existence check.